packages/react/src/components/PageHeader/PageHeader.tsx

```diff
@@ -241,6 +235,176 @@ PageHeaderContent.propTypes = {
   pageActions: PropTypes.node,
 };
 
+/**
+ * ----------------
+ * PageHeaderContentPageActions
+ * ----------------
+ */
+interface PageHeaderContentPageActionsProps {
+  /**
+   * Provide child elements to be rendered inside PageHeaderContentPageActions.
+   */
+  children?: React.ReactNode;
+  /**
+   * Specify an optional className to be added to your PageHeaderContentPageActions
+   */
+  className?: string;
+  /**
+   *  The PageHeaderContent's page actions collabpsible Menu button label
+   */
+  menuButtonLabel?: string;
+  /**
+   * The PageHeaderContent's page actions
+   */
+  pageActions?: React.ReactNode;
+}
+const PageHeaderContentPageActions = React.forwardRef<
+  HTMLDivElement,
+  PageHeaderContentPageActionsProps
+>(
+  ({
+    className,
+    children,
+    menuButtonLabel = 'Actions',
+    pageActions,
+    ...other
+  }: PageHeaderContentPageActionsProps) => {
+    const prefix = usePrefix();
+    const classNames = classnames(
+      {
+        [`${prefix}--page-header__content__page-actions`]: true,
+      },
+      className
+    );
+    const containerRef = useRef<HTMLDivElement>(null);
+    const offsetRef = useRef<HTMLDivElement>(null);
+    const [menuButtonVisibility, setMenuButtonVisibility] = useState(false);
+    const [hiddenItems, setHiddenItems] = useState<any[]>([]);
+
+    // need to set the grid columns width based on the menu button's width
+    // to avoid overlapping when resizing
+    useLayoutEffect(() => {
+      if (menuButtonVisibility && offsetRef.current) {
+        const width = offsetRef.current.offsetWidth;
+        document.documentElement.style.setProperty(
+          '--pageheader-title-grid-width',
+          `${width}px`
+        );
+      }
+    }, [menuButtonVisibility]);
+
+    useEffect(() => {
+      if (!containerRef.current || !Array.isArray(pageActions)) return;
+      createOverflowHandler({
+        container: containerRef.current,
+        // exclude the hidden menu button from children
+        maxVisibleItems: containerRef.current.children.length - 1,
+        onChange: (visible, hidden) => {
+          setHiddenItems(pageActions?.slice(visible.length));
+
+          if (hidden.length > 0) {
+            setMenuButtonVisibility(true);
+          }
+        },
+      });
+    }, []);
+
+    return (
+      <div className={classNames} ref={containerRef}>
+        {pageActions && (
+          <>
+            {Array.isArray(pageActions) && (
+              <>
+                {pageActions.map((action) => (
+                  <div key={action.id} className="action">
+                    {action.body}
+                  </div>
+                ))}
+                <span data-offset data-hidden ref={offsetRef}>
+                  <MenuButton
+                    menuAlignment="bottom-end"
+                    label={menuButtonLabel}
+                    size="md">
+                    {hiddenItems.reverse().map((item) => (
```

Comment:
Suggestion to avoid mutating `hiddenItems`:

```suggestion
                    {[...hiddenItems].reverse().map((item) => (
```